src/libraries/System.Private.CoreLib/src/System/Environment.cs

```diff
@@ -105,7 +105,14 @@ public static string ExpandEnvironmentVariables(string name)
             s_commandLineArgs = cmdLineArgs;
         }
 
-        public static string GetFolderPath(SpecialFolder folder) => GetFolderPath(folder, SpecialFolderOption.None);
+// We don't expect most of these folders to exist on Android by default
+#if TARGET_ANDROID
+        private static SpecialFolderOption defaultSpecialFolderOption = SpecialFolderOption.DoNotVerify;
+#else
+        private static SpecialFolderOption defaultSpecialFolderOption = SpecialFolderOption.None;
+#endif
```

Comment:
A few things:
1. This should be a const with PascalCased naming rather than a (non-readonly) static field.
2. Environment uses partial files rather than ifdefs to split its functionality across platforms.
3. I'm not clear why we want this default behavior to diverge for Android, anyway.